src/Cake.Common.Tests/Unit/IO/DirectoryAliasesTests.cs

```diff
@@ -196,6 +196,48 @@ public void Should_Create_Directory_If_Missing()
                 // Then
                 Assert.True(fixture.FileSystem.Exist((DirectoryPath)"/NonExisting"));
             }
+
+            [Fact]
+            public void Should_Skip_Predicate_Folder()
+            {
+                // Given
+                var directory = new DirectoryPath("/Temp");
+                var fixture = new FileSystemFixture();
+                var context = Substitute.For<ICakeContext>();
+                context.FileSystem.Returns(fixture.FileSystem);
+                var preHelloFiles = context.FileSystem
+                    .GetDirectory("/Temp/Hello/")
+                    .GetFiles("*.*", SearchScope.Recursive)
+                    .Select(file=>file.Path.FullPath)
+                    .ToArray();
+
+                var preGoodbyeFiles = context.FileSystem
+                    .GetDirectory("/Temp/Goodbye/")
+                    .GetFiles("*.*", SearchScope.Recursive)
+                    .Select(file=>file.Path.FullPath)
+                    .ToArray();
+
+                // When
+                context.CleanDirectory(directory, predicate=>predicate.Path.FullPath!="/Temp/Goodbye");
```

Comment:
Use the `DirectoryAliases` type instead of calling the extension method to make it clear what's under test.